detection_rules/rule.py

```diff
@@ -966,17 +985,32 @@ def get_version_space(self) -> Optional[int]:
 
     @property
     def latest_version(self) -> Optional[int]:
-        """Retrieve the latest known version of the rule."""
-        min_stack = self.get_supported_version()
-        return self.version_lock.get_locked_version(self.id, min_stack)
+        """Retrieve the latest known version of the rule, considering the version lock setting."""
+        toml_version = self.data.get("version")
+
+        if BYPASS_VERSION_LOCK:
+            return toml_version if toml_version else 1
+
+        if toml_version:
+            print(f"WARNING: Rule {self.name} - {self.id} has a version set in the rule TOML."
+                  " This `version` will be ignored and defaulted to the version.lock.json file."
+                  " Set `bypass_version_lock` to `True` in the rules config to use the TOML version.")
+
+        return self.version_lock.get_locked_version(self.id, self.get_supported_version())
 
     @property
     def autobumped_version(self) -> Optional[int]:
         """Retrieve the current version of the rule, accounting for automatic increments."""
         version = self.latest_version
+
+        if BYPASS_VERSION_LOCK:
+            return version
```

Comment:
should we raise an error here instead?

Reply:
I see why but I think it's safe to leave this since the field is used in other places not really related to version lock

Reply:
my concern is that `latest_version` != `data.version` != `autobumped_version` and so it is misleading to reuse this method for convenience just because it used in a lot of places. This should only be called when there is an expectation of a bump with detected changes.

Reply:
The field is inconsistently used, so we should probably fix this in main and not this PR.

Reply:
looking at all the calls it doesn't seem to be inconsistent. In main there is _only_ the concept of a version locked approach, so it _always_ make sense to call this to retrieve the _true_ latest version